vpr/src/base/vpr_constraints.h

```diff
@@ -0,0 +1,52 @@
+#ifndef VPR_CONSTRAINTS_H
+#define VPR_CONSTRAINTS_H
+
+#include "vtr_vector.h"
+#include "vpr_utils.h"
+#include "partition.h"
+#include "partition_regions.h"
+
+/**
+ * @file
+ * @brief This file defines the VprConstraints class used to store and read out data related to user-specified
+ * 		  block and region constraints for the packing and placement stages.
+ *
+ * Overview
+ * ========
+ * This class contains functions that read in and store information from a constraints XML file.
+ * The XML file provides a partition list, with the names/ids of the partitions, and each atom in the partition.
+ * It also provides placement constraints, where it is specified which regions the partitions should go in.
+ * In the constraints file, a placement constraint can also be created for a specific primitive. A partition would be automatically
+ * created for the primitive when a placement constraint was specified for it.
+ *
+ */
+
+class VprConstraints {
+  public:
+    /**
+     * Returns the intersection of two PartitionRegions vectors that are passed to it.
+     * Can be used by atom_cluster_intersection.
+     */
+    PartitionRegions do_regions_intersect(PartitionRegions part_regions_old, PartitionRegions part_regions_new);
```

Comment:
Rename to get_intersection.
Make a member function of PartitionRegions.
part_regions_old and new don't seem like good names.  region1 and region2 would be better.  Once a member function, you just need one (the other region is the member variables of the class instance).

You probably also want a test for intersection.  Could be a member function:
bool intersects (PartitionRegion region)

Reply:
Objects you are not going to change that are passed in are best passed in as const &  (constant references).